pkg/hds/server/server.go

```diff
@@ -136,7 +139,7 @@ func (s *server) process(stream Stream, reqOrRespCh chan *envoy_service_health.H
 				TypeUrl:       envoy_cache.HealthCheckSpecifierType,
 				ResourceNames: []string{"hcs"},
 				VersionInfo:   lastVersion,
-			}, responseChan)
+			}, streamState, responseChan)
```

Comment:
maybe it's better to create a new streamState for each `CreateWatch`, the name suggests this object may have some state in it